src/devtools/devtools.tsx

```diff
@@ -378,9 +407,16 @@ export const ReactQueryDevtoolsPanel = React.forwardRef(
 
     React.useEffect(() => {
       if (isOpen) {
-        return queryCache.subscribe(() => {
+        const unsubscribe = queryCache.subscribe(() => {
           setUnsortedQueries(Object.values(queryCache.getAll()))
         })
+        // re-subscribing after the panel is closed and re-opened won't trigger the callback,
+        // So we'll manually populate our state
+        setUnsortedQueries(Object.values(queryCache.getAll()))
+
+        return () => {
+          unsubscribe()
+        }
```

Comment:
would this also work?

```suggestion
        return unsubscribe
```

Reply:
Yes, but it would do the same thing at the cost of another function declaration, right?

Reply:
quite the contrary. The previous solution declares a new inline arrow function and returns that, while the proposed solution just returns the already declared function _unsubscribe_ for react to invoke. It doesn't matter perf wise so we can also leave it as is :)

Reply:
Oh sorry, I was reading this on my phone and it looked backwards for some reason, lol